### PR TITLE
chore: Enable `noUncheckedIndexedAccess`

### PR DESCRIPTION
**Pull request by akash1810**
It might be easier to review this change commit by commit.

## What does this change?

[Introduced](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#checked-indexed-accesses---nouncheckedindexedaccess) in TS 4.1, [`noUncheckedIndexedAccess`](https://www.typescriptlang.org/tsconfig#noUncheckedIndexedAccess) offers stricter type checking, which is a good thing!

From the release notes:

> That means that in our last example, `opts.yadda` will have the type `string | number | undefined` as opposed to just `string | number`. If you need to access that property, you’ll either have to check for its existence first or use a non-null assertion operator (the postfix `!` character).

## How to test

CI passing should be sufficient.

### CODE REVIEW

File: src/bin/commands/new-project/utils/imports.ts [hunk 1]
```diff
@@ -8,33 +8,66 @@ interface Import {
 }
 
 export class Imports {
-  imports: Record<string, Import>;
+  private readonly imports: Record<string, Import>;
 
-  constructor(imports?: Record<string, Import>) {
-    this.imports = imports ?? {};
+  private constructor(imports: Record<string, Import>) {
```
Comment (akash1810) [inline src/bin/commands/new-project/utils/imports.ts:13]: Although `Imports` is exported, instantiation is only performed by:
  - `newStackImports`
  - `newAppImports`
  - `newTestImports`

In this PR we make `Imports`'s constructor private, and convert the aforementioned functions to static methods on the class to improve namespacing.

File: src/bin/commands/new-project/utils/imports.ts [hunk 2]
```diff
@@ -8,33 +8,66 @@ interface Import {
...
+    this.imports = imports;
   }
 
-  addImport(lib: string, components: string[], type = false): void {
```
Comment (akash1810) [inline src/bin/commands/new-project/utils/imports.ts:17]: Calls to `addImport` always had `type` set to `false`. This meant we could remove the `type` argument. Once that's done, the value of `addImport` quickly disappears, so we can just remove it!

File: src/constants/ssm-parameter-paths.ts [hunk 3]
```diff
@@ -52,3 +64,7 @@ export const SSM_PARAMETER_PATHS: Record<string, SsmParameterPath> = {
     optional: true,
   },
 };
+
+export const ALL_SSM_PARAMETER_PATHS: SsmParameterPath[] = Object.values(
+  NAMED_SSM_PARAMETER_PATHS
+) as SsmParameterPath[];
```
Comment (akash1810) [inline src/constants/ssm-parameter-paths.ts:70]: Keeping things DRY here.

File: src/constants/ssm-parameter-paths.ts [hunk 4]
```diff
@@ -8,9 +8,21 @@ export interface SsmParameterPath {
...
+  PrimaryVpcId: SsmParameterPath;
+  PrimaryVpcPrivateSubnets: SsmParameterPath;
+  PrimaryVpcPublicSubnets: SsmParameterPath;
+  FastlyCustomerId: SsmParameterPath;
+}
+
 export const VPC_SSM_PARAMETER_PREFIX = "/account/vpc";
 
-export const SSM_PARAMETER_PATHS: Record<string, SsmParameterPath> = {
+export const NAMED_SSM_PARAMETER_PATHS: NamedSsmParameterPaths = {
```
Comment (akash1810) [inline src/constants/ssm-parameter-paths.ts:25]: `NAMED_SSM_PARAMETER_PATHS` is slightly clearer than `SSM_PARAMETER_PATHS`?

**Review by jacobwinch — APPROVED** [review]
👍

**Pull request merged**